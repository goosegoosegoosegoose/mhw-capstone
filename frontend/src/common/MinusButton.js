import React, { useState, useContext, useEffect } from "react";
import UserContext from "../auth/userContext";
import { Button } from "react-bootstrap";

const MinusButton = ({id, spacing, minus}) => {
  const currentUser = useContext(UserContext);
  const [exists, setExists] = useState(null);

  useEffect(() => {
    if (currentUser.decorations) {
      setExists(currentUser.decorations.hasOwnProperty(id));
    }
  }, [currentUser.decorations])

  const handleMinus = () => {
    minus(id)
  }
  
  if (!currentUser) {
    return (
      <Button className={spacing} variant="danger" size="sm" onClick={handleMinus} disabled>-1</Button>
    )
  }

  return (
    <>
      {exists ? <Button className={spacing} variant="danger" size="sm" onClick={handleMinus}>-1</Button>
      : <Button className={spacing} variant="danger" size="sm" onClick={handleMinus} disabled>-1</Button>}
    </>
  )
}

export default MinusButton