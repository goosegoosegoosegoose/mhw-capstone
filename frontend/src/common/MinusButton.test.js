import { render } from '@testing-library/react';
import MinusButton from "./MinusButton";

test('smoke test', () => {
  render(      
    <MinusButton />
  );
});

test('snapshot', () => {
  const { asFragment } = render(<MinusButton />);
  expect(asFragment()).toMatchSnapshot();
});