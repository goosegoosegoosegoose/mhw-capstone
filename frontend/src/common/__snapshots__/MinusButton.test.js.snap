// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn btn-danger btn-sm"
    disabled=""
    type="button"
  >
    -1
  </button>
</DocumentFragment>
`;
